tools/auth/wakatime.go

```diff
@@ -0,0 +1,83 @@
+package auth
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/pocketbase/pocketbase/tools/types"
+	"golang.org/x/oauth2"
+)
+
+func init() {
+	Providers[NameWakatime] = wrapFactory(NewWakatimeProvider)
+}
+
+var _ Provider = (*Wakatime)(nil)
+
+// NameWakatime is the unique name of the Wakatime provider.
+const NameWakatime = "wakatime"
+
+// Wakatime is an auth provider for Wakatime.
+type Wakatime struct {
+	BaseProvider
+}
+
+// NewWakatimeProvider creates a new Wakatime provider instance with some defaults.
+func NewWakatimeProvider() *Wakatime {
+	return &Wakatime{BaseProvider{
+		ctx:         context.Background(),
+		displayName: "Wakatime",
+		pkce:        false,
```

Comment:
Try to change the default `pkce` option to `true`, unless Wakatime explicitly requires to be disabled _(so far only LinkedIn OIDC has been reported to have this quirk which is also the reason why the field was added in the first place)_.

In general most providers usually will just ignore the extra PKCE query parameters if they don't support it.